From: TNG Technical Manual compatibility investigation
Subject: [PATCH 2/4] DIRECTOR: Restart digital video assigned with setCast

Reset timing and restart a digital video whenever Lingo assigns it to a sprite,
including reassignment of the same cast number after its filename changed.
---
 engines/director/channel.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/engines/director/channel.cpp b/engines/director/channel.cpp
index 8f53c3dc..be425476 100644
--- a/engines/director/channel.cpp
+++ b/engines/director/channel.cpp
@@ -484,16 +484,20 @@ void Channel::setCast(CastMemberID memberID) {
 
 	// Duplicate of the special cases in setClean.
 	// Maybe it makes sense to force setClean to use setCast instead?
-	if (hasChanged && _sprite->_cast) {
+	if (_sprite->_cast) {
 		if (_sprite->_cast->_type == kCastDigitalVideo) {
 			DigitalVideoCastMember *dv = (DigitalVideoCastMember *)_sprite->_cast;
 			if (dv->loadVideoFromCast()) {
 				_movieTime = 0;
+				_movieRate = 1.0;
 				dv->setChannel(this);
+				_startTime = 0;
+				_stopTime = dv->getMovieTotalTime();
+				dv->rewindVideo();
 				dv->startVideo();
 			}
-		} else if (_sprite->_cast->_type == kCastFilmLoop ||
-					_sprite->_cast->_type == kCastMovie) {
+		} else if (hasChanged && (_sprite->_cast->_type == kCastFilmLoop ||
+					_sprite->_cast->_type == kCastMovie)) {
 			// brand new film loop, reset the frame counter.
 			_filmLoopFrame = 1;
 		}
-- 
2.51.0
